grammar/Python.g

```diff
@@ -488,7 +488,7 @@ decorator
       RPAREN
     |
       {
-          $etype = $dotted_attr.etype;
+          $etype = actions.makeCall($LPAREN, $dotted_attr.etype);
```

Comment:
Use of `LPAREN` seems dubious here as we have not encountered a match of that kind.

Since `dotted_attr` is able to match either a simple name or an attribute expression (and returns a different AST node type in each case) I am beginning to think the problem might be in the processing of the decorators subsequently, that it overlooks the possibility of an attribute expression.

Reply:
> Use of `LPAREN` seems dubious here as we have not encountered a match of that kind.

@tatianacv can correct me if I am mistaken, but I think she just needed to send *something* in as an argument there.

Reply:
> Since `dotted_attr` is able to match either a simple name or an attribute expression (and returns a different AST node type in each case) I am beginning to think the problem might be in the processing of the decorators subsequently, that it overlooks the possibility of an attribute expression.

I believe that @tatianacv is modeling this change off of the [case where there is parentheses](https://github.com/jython/jython/pull/274/files#diff-ff477a8df29d93b68543d917d4e9b0e31d828d9e218cae41e10774443a20d5e2L480-L481). The idea, if I am not mistaken, is that, with decorators (as opposed to "normal" function calls), the following is semantically equivalent:

```python
@tf.function
def func():
    pass

@tf.function()
def func2():
    pass
```

Thus, they should be processed the same. That is what we are seeing in our client code, and that is the fixed being proposed. I agree that the argument is dubious, but I believe that there is no `makeCall()` that takes a single parameter. Perhaps an overloaded `makeCall()` should be part of this PR.

Reply:
> > Use of `LPAREN` seems dubious here as we have not encountered a match of that kind.
> 
> @tatianacv can correct me if I am mistaken, but I think she just needed to send _something_ in as an argument there.

It looks like `$LPAREN` would be `null` in practice. That may be tolerated in later processing (perhaps not used unless there is a syntax error in need of a location) but is ... dubious.

Reply:
@tatianacv Could you confirm that there is no `makeCall()` that takes only a single parameter that would match the argument `$dotted_attr.etype`?